config/config.go

```diff
@@ -21,34 +21,55 @@
 package config
 
 import (
-	"github.com/pkg/errors"
-	zanzibar "github.com/uber/zanzibar/runtime"
+	"os"
+
+	"github.com/uber/zanzibar/runtime"
 )
 
-// NewRuntimeConfigOrDie returns a static config struct that is pre-set with
-// the service configuration defaults
+// EnvConfig map from environment variable to config key and data type
+type EnvConfig map[string]struct {
+	Key      string `json:"key"`
+	DataType string `json:"dataType"`
+}
+
+// NewRuntimeConfigOrDie returns a static config struct
+// that is pre-set with the service configuration defaults
+// and overridden by service.env.config
 func NewRuntimeConfigOrDie(
 	files []string,
 	seedConfig map[string]interface{},
 ) *zanzibar.StaticConfig {
 	defaultConfig, err := defaultConfig()
 	if err != nil {
-		panic(errors.Wrap)
+		panic("error getting default config")
 	}
-	serviceConfig := make([]*zanzibar.ConfigOption, len(files)+1)
-	serviceConfig[0] = defaultConfig
-	for i, configFilePath := range files {
-		serviceConfig[i+1] = zanzibar.ConfigFilePath(configFilePath)
+
+	var serviceConfig []*zanzibar.ConfigOption
+	serviceConfig = append(serviceConfig, defaultConfig)
+	for _, configFilePath := range files {
+		serviceConfig = append(serviceConfig, zanzibar.ConfigFilePath(configFilePath))
 	}
 
-	return zanzibar.NewStaticConfigOrDie(serviceConfig, seedConfig)
+	staticConfig := zanzibar.NewStaticConfigOrDie(serviceConfig, seedConfig)
+	getEnvConfig(staticConfig)
+
+	return staticConfig
 }
 
 func defaultConfig() (*zanzibar.ConfigOption, error) {
 	bytes, err := Asset("production.json")
 	if err != nil {
 		return nil, err
 	}
-
 	return zanzibar.ConfigFileContents(bytes), nil
 }
+
+func getEnvConfig(cfg *zanzibar.StaticConfig) {
```

Comment:
the function shouldn't be called "get" if it doesn't return anything, "setEnvConfig" would be better